scaleway/resource_vpc_private_network.go

```diff
@@ -131,17 +142,20 @@ func resourceScalewayVPCPrivateNetworkDelete(ctx context.Context, d *schema.Reso
 		return diag.FromErr(err)
 	}
 
-	err = vpcAPI.DeletePrivateNetwork(&vpc.DeletePrivateNetworkRequest{
-		PrivateNetworkID: ID,
-		Zone:             zone,
-	}, scw.WithContext(ctx))
-
-	if err != nil && !is404Error(err) {
-		if is412Error(err) {
-			// TODO: manage this error
-			l.Warningf("error deleting private network in zone (%s): %s", zone, err)
-			return nil
+	err = retryVPCContext(ctx, retryGWTimeout, func() *resource.RetryError {
+		errPN := vpcAPI.DeletePrivateNetwork(&vpc.DeletePrivateNetworkRequest{
+			PrivateNetworkID: ID,
+			Zone:             zone,
+		}, scw.WithContext(ctx))
+		if errPN != nil {
+			if is409Error(errPN) || is412Error(errPN) {
+				return resource.RetryableError(errPN)
```

Comment:
Can this go into an infinite loop?

Reply:
The loop is limited by a timeout and the client allow 4  attempts